index.html

```diff
@@ -295,20 +330,45 @@ <h2>
           input device SHOULD appear next to each other in the axes array, X
           then Y. It is RECOMMENDED that axes appear in decreasing order of
           importance, such that element 0 and 1 typically represent the X and Y
-          axis of a directional stick. The same object MUST be returned until
-          the <a>user agent</a> needs to return different values (or values in
-          a different order).
+          axis of a directional stick.
+          </p>
+          <p>
+          When the system receives new data from a connected gamepad, a
+          microtask MUST be queued with the user interaction task source to
+          update the gamepad state. If the gamepad has the same axes in the same
+          order as the previous update, and if no axis values have changed
+          since the previous update, then the axes attribute MUST return the
+          same array object. If the number of axes has changed, or if any axis
+          value has changed, then the axis attribute MUST return a new array.
+          </p>
+          <p>
+          Once the system is notified that a previously connected gamepad has
+          been disconnected, the axes attribute MUST continue to return the
+          last state received from the gamepad before disconnection.
+          </p>
         </dd>
         <dt>
           <dfn>buttons</dfn> attribute
         </dt>
         <dd>
-          Array of button states for all buttons of the gamepad. It is
-          RECOMMENDED that buttons appear in decreasing importance such that
-          the primary button, secondary button, tertiary button, and so on
-          appear as elements 0, 1, 2, ... in the buttons array. The same object
-          MUST be returned until the <a>user agent</a> needs to return
-          different values (or values in a different order).
+          <p>
+          Array of {{GamepadButton}} objects representing the current state of
+          all buttons of the gamepad. It is RECOMMENDED that buttons appear in
+          decreasing importance such that the primary button, secondary
+          button, tertiary button, and so on appear as elements 0, 1, 2, ...
+          in the buttons array.
+          </p>
+          <p>
+          When the system receives new data from a connected gamepad, a
+          microtask MUST be queued with the user interaction task source to
+          update the gamepad state. If the gamepad has the same buttons in the
+          same order as the previous update, the buttons attribute MUST return
+          the same array object. If the number of buttons or the ordering of
+          buttons has changed, the buttons attribute MUST return a new array
```

Comment:
This suggests that we can't make Gamepad a live object and allow the number of buttons or axes to change. This would diverge from the original spec, but is probably not important since I don't think there are any implementations that support this behavior. I'll remove the language about changes to the length of the buttons and axes arrays.

Reply:
This would work for the buttons attribute but I'm not sure how to resolve it for axes. The axes attribute is FrozenArray<double> so we can't change the values in-place like we can for the buttons array.

I tested the current behavior in Firefox. `pad.axes === pad.axes` always returns true, but if you save the axes array before and after interacting with the joystick then you'll get different arrays:

```
let pad = navigator.getGamepads()[0];
let axes = pad.axes;
axes === pad.axes; // true
// wiggle the thumbstick
axes === pad.axes; // false
```

Reply:
argh... that's not good. @domenic, sorry to drag you in... if you have some cycles, could you suggest how we might proceed? This is a bit a of a pickle :( 

We might need an array-like object here instead 🤢

Reply:
The behavior defined in the previous comment seems right to me. There is an async turn boundary between the value of the property changing which is very reasonable.

Also, I suspect both buttons and axes should have this behavior, not just axes.

Reply:
I guess that make sense... it doesn't feel good to swap out arrays, but yes, the async/looping aspects does make it reasonable. Thank you.